service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -805,16 +806,27 @@ public void createResourceFailure(
       String flightId,
       @Nullable Exception exception,
       WsmResourceStateRule resourceStateRule) {
-
+    DbResource dbResource =
+        getDbResourceFromIds(resource.getWorkspaceId(), resource.getResourceId());
     switch (resourceStateRule) {
       case DELETE_ON_FAILURE -> {
-        deleteResourceWorker(
-            resource.getWorkspaceId(), resource.getResourceId(), /*resourceType=*/ null);
+        // There is no guarantee this is the flight which created this resource. Validate that it
+        // is before attempting to delete the workspace.
+        try {
+          stateDao.updateState(
```

Comment:
An alternative would be to always make the state change to BROKEN. Then if it is DELETE_ON_FAILURE, do the delete. 

I am thinking we want to do the catch in both cases to avoid throwing a new error that Stairway would see as a dismal failure. Right?

Reply:
Good point, we need to catch in both cases to avoid that error message.

IMO using the switch statement and setting the state to `NOT_EXISTS` for `DELETE_ON_FAILURE` is more verbose but worthwhile since it's easier to follow, agreed they're functionally equivalent though.